builtin/k8s/platform.go

```diff
@@ -296,318 +294,392 @@ func (p *Platform) resourceDeploymentStatus(
+		envVars[k] = v
+	}
+	for k, v := range deployConfig.Env() {
+		envVars[k] = v
+	}
+
+	// Check autoscaling
+	if p.config.AutoscaleConfig != nil && p.config.CPU == nil {
+		ui.Output("For autoscaling in Kubernetes to work, a deployment must specify "+
+			"cpu resource limits and requests. Otherwise the metrics-server will not properly be able "+
+			"to scale your deployment.", terminal.WithWarningStyle())
+	}
+
+	// Create scratch space volumes
+	var volumes []corev1.Volume
+	for idx := range p.config.ScratchSpace {
+		scratchName := fmt.Sprintf("scratch-%d", idx)
+		volumes = append(volumes,
+			corev1.Volume{
+				Name: scratchName,
+				VolumeSource: corev1.VolumeSource{
+					EmptyDir: &corev1.EmptyDirVolumeSource{},
 				},
 			},
-			InitialDelaySeconds: initialDelaySeconds,
-			TimeoutSeconds:      timeoutSeconds,
-			FailureThreshold:    failureThreshold,
-		}
+		)
 	}
 
-	if len(p.config.ScratchSpace) > 0 {
-		for idx, scratchSpaceLocation := range p.config.ScratchSpace {
-			scratchName := fmt.Sprintf("scratch-%d", idx)
-			deployment.Spec.Template.Spec.Volumes = append(
-				deployment.Spec.Template.Spec.Volumes,
-				corev1.Volume{
-					Name: scratchName,
-					VolumeSource: corev1.VolumeSource{
-						EmptyDir: &corev1.EmptyDirVolumeSource{},
-					},
-				},
-			)
+	// Configure containers
+	var command *[]string
+	var args *[]string
+	if p.config.Pod != nil && p.config.Pod.Container != nil {
+		command = p.config.Pod.Container.Command
+		args = p.config.Pod.Container.Args
+	}
+
+	appContainer, err := configureK8sContainer(
+		src.App,
+		fmt.Sprintf("%s:%s", img.Image, img.Tag),
+		p.config.Ports,
+		envVars,
+		p.config.Probe,
+		p.config.ProbePath,
+		p.config.CPU,
+		p.config.Memory,
+		p.config.Resources,
+		command,
+		args,
+		p.config.ScratchSpace,
+		volumes,
+		log,
+	)
+	if err != nil {
+		return status.Errorf(status.Code(err),
+			"Failed to define app container: %s", err)
+	}
 
-			deployment.Spec.Template.Spec.Containers[0].VolumeMounts = append(
-				deployment.Spec.Template.Spec.Containers[0].VolumeMounts,
-				corev1.VolumeMount{
-					Name:      scratchName,
-					MountPath: scratchSpaceLocation,
-				},
+	var sidecarContainers []corev1.Container
+	if p.config.Pod != nil {
+		for _, sidecarConfig := range p.config.Pod.Sidecars {
+			sidecarEnvVars := make(map[string]string)
+			// Add deploy config environment to container env vars
+			for k, v := range sidecarConfig.StaticEnvVars {
+				sidecarEnvVars[k] = v
+			}
+			for k, v := range deployConfig.Env() {
+				sidecarEnvVars[k] = v
+			}
+
+			sidecarContainer, err := configureK8sContainer(
+				sidecarConfig.Name,
+				sidecarConfig.Image,
+				sidecarConfig.Ports,
+				sidecarConfig.StaticEnvVars,
+				sidecarConfig.Probe,
+				sidecarConfig.ProbePath,
+				sidecarConfig.CPU,
+				sidecarConfig.Memory,
+				sidecarConfig.Resources,
+				sidecarConfig.Command,
+				sidecarConfig.Args,
+				p.config.ScratchSpace,
+				volumes,
+				log,
```

Comment:
Same here, we can pass through a `Container` struct and simplify this (it should already be a Container struct so we won't have to create it again like we would for the app container earlier)